include/bout/region.hxx

```diff
@@ -157,9 +194,55 @@ public:
 
   /// Modulus operator
   SpecificInd<N> operator%(int n) {
-    SpecificInd<N> new_ind(ind % n);
+    SpecificInd<N> new_ind{ind % n, ny, nz};
     return new_ind;
   }
+
+  /// Convenience functions for converting to (x, y, z)
+  int x() const { return (ind / nz) / ny; }
+  int y() const { return (ind / nz) % ny; }
+  int z() const { return (ind % nz); }
+
+  const inline SpecificInd xp(int dx = 1) const { return {ind + (dx * ny * nz), ny, nz}; }
```

Comment:
I think all these need `<N>` adding?

Reply:
Actually, I think it's the other way round -- we don't need to mention `<N>` anywhere inside the class definition because we don't actually use it. The whole class has been templated on `N`.

Likewise, I think we don't need to write `Region<T>` inside the `Region` class definition

Reply:
Ah ok, I thought the return types would require knowledge of what type to template on but I didn't test this was the case.